atomic_reactor/plugins/pre_pull_base_image.py

```diff
@@ -37,17 +37,29 @@ def __init__(self, tasker, workflow, parent_registry=None, parent_registry_insec
         self.parent_registry = parent_registry
         self.parent_registry_insecure = parent_registry_insecure
 
-    def run(self):
-        """
-        pull base image
-        """
-        base_image = self.workflow.builder.base_image
+    def resolve_base_image(self, build_json):
+        spec = build_json.get("spec")
+        try:
+            image_id = spec['triggeredBy'][0]['imageChangeBuild']['imageID']
+            base_image = ImageName.parse(image_id)
+        except (TypeError, KeyError, IndexError):
+            base_image = self.workflow.builder.base_image
         if self.parent_registry is not None:
             self.log.info("pulling base image '%s' from registry '%s'",
                           base_image, self.parent_registry)
         else:
             self.log.info("pulling base image '%s'", base_image)
```

Comment:
I feel like these log messages belong in `run()` after this method has returned. Perhaps these should be merged into the `if self.parent_registry: // else:` that's already there.

Reply:
fixed.